build/media-suite_helper.sh

```diff
@@ -1379,16 +1379,37 @@ log() {
 }
 
 create_build_dir() {
-    local build_dir="build${1:+-$1}-$bits"
-    if [[ ! -f "$(get_first_subdir)/do_not_clean" ]]; then
-        if [[ "$(basename "$(pwd)")" == "$build_dir" ]]; then
-            rm -rf ./* && cd_safe ".."
-        elif [[ -d $build_dir ]] && ! rm -rf ./"$build_dir"; then
-            cd_safe "$build_dir" && rm -rf ./* && cd_safe ".."
-        fi
+    local print_build_dir=false nocd=${nocd:-false} norm=false build_root repo_dir build_dir getoptopt OPTARG OPTIND
+    build_root="$LOCALBUILDDIR/$(get_first_subdir)"
+    while getopts ":pcrC:" getoptopt; do
+        case $getoptopt in
+        p) print_build_dir=true ;;
+        c) nocd=true ;;
+        r) norm=true ;;
+        C) build_root="$OPTARG" ;;
+        \?)
+            echo "Invalid Option: -$OPTARG" 1>&2
+            return 1
+            ;;
+        :)
+            echo "Invalid option: $OPTARG requires an argument" 1>&2
+            return 1
+            ;;
+        esac
+    done
+    shift $((OPTIND - 1))
+
+    build_dir="$build_root/build${1:+-$1}-$bits"
+
+    if [[ -d $build_dir ]]; then
+        [[ ! -f $LOCALBUILDDIR/$(get_first_subdir)/do_not_clean ]] &&
```

Comment:
Shouldn't it be an OR instead?

Reply:
No since it should first check if `do_not_clean` does not exist, then if norm is false, then try to remove the files

Reply:
I think

Reply:
But you need do_not_clean to be there so the option works. If the file doesn't exist the option does nothing. Isn't that redudant?

Reply:
If you always need to do two things to do a specific action, but if you forget one of them, then it should be a single thing anyway.

Reply:
I'll check the logic again, I'm confusing myself

Reply:
Switched around the logic

Reply:
Looks more or less the same. Don't you still need do_not_clean for the option to do anything?

Reply:
No, as if `do_not_clean` exists, create_build_dir should not delete any files and not do anything, so it should skip the whole block if it exists

Reply:
Right.